src/linear_algebra/free_module/strong_rank_condition.lean

```diff
@@ -43,7 +43,9 @@ begin
   { rwa strong_rank_condition_iff_succ R },
   intros n f, by_contradiction hf,
 
-  letI := module.finite.of_basis (pi.basis_fun R (fin (n + 1))),
+  -- Lean is unable to find this instance without help, either via this `letI`, or via a duplicate
+  -- instance with unecessarily strong typeclasses on `R` and `M`.
+  letI : module.finite R (fin n.succ → R) := module.finite.pi,
```

Comment:
This seems like the same problem I was experiencing in #8221, where typeclass search doesn't try hard enough to find the right `semiring R` and `add_comm_monoid` arguments that match the `module` argument it found. Rather than adding a pile of global instances just to be safe, my hope is lean4 will solve this.